### PR TITLE
Fix: adjust code for not recognized ison codes

### PR DESCRIPTION
**Pull request by martinig94**
Filtering out unrecognized iso3 codes which are for now hardcoded in a vector.
Ideally these could be stored in a table of a database

### CODE REVIEW

**Comment by zackarno** [comment]
looks like a nice workaround patch, but currently getting this error:

```
df_wrangled <- wrangle_conflict$wrangle(df_raw)
#> Error in `reframe()`:
#> ℹ In argument: `complete(data = pick(everything()), ..., fill = fill, explicit = explicit)`.
#> ℹ In group 236: `iso3 = NA` and `acled_hdx_url = NA`.
#> Caused by error in `seq.int()`:
#> ! 'to' must be a finite number
#> Run `rlang::last_trace()` to see where the error occurred.
#> Warning message:
#> There was 1 warning in `dplyr$mutate()`.
#> ℹ In argument: `iso3 = location_codes$ison_to_iso3(as.numeric(iso))`.
#> Caused by warning:
#> ! Some values were not matched unambiguously: 2 
```

**Comment by zackarno** [comment]
> looks like a nice workaround patch, but currently getting this error:
> 
> ```
> df_wrangled <- wrangle_conflict$wrangle(df_raw)
> #> Error in `reframe()`:
> #> ℹ In argument: `complete(data = pick(everything()), ..., fill = fill, explicit = explicit)`.
> #> ℹ In group 236: `iso3 = NA` and `acled_hdx_url = NA`.
> #> Caused by error in `seq.int()`:
> #> ! 'to' must be a finite number
> #> Run `rlang::last_trace()` to see where the error occurred.
> #> Warning message:
> #> There was 1 warning in `dplyr$mutate()`.
> #> ℹ In argument: `iso3 = location_codes$ison_to_iso3(as.numeric(iso))`.
> #> Caused by warning:
> #> ! Some values were not matched unambiguously: 2 
> ```

wait! i just restarted my R-session and the error went away, my bad! classic...

File: src/utils/location_codes.R [hunk 1]
```diff
@@ -37,10 +37,13 @@ names_to_iso3 <- function(location_names) {
...
+  not_recognized_ison <- c(2)
   iso3 <- character(length(ison))
   iso3[ison == 0] <- "XKX"
-  iso3[ison != 0] <- countrycode$countrycode(
-    sourcevar = ison[ison != 0],
+  iso3[! ison %in% c(not_recognized_ison, 0)] <- countrycode$countrycode(
+    sourcevar = ison[! ison %in% c(not_recognized_ison, 0)],
     origin = "iso3n",
     destination = "iso3c"
   )
```
Comment (zackarno) [inline src/utils/location_codes.R:49]: this works! it looks like Seth assigned `XKX` to `ison == 0` but then disregards that country code when converting back to `iso3c`, but then he deals with `XKX` and other custom names in the end with

https://github.com/OCHA-DAP/hdx-signals/blob/37fd8ab78787c40c7ad93911a741be306e4e52f7/src/utils/location_codes.R#L13-L32

**Review by zackarno — APPROVED** [review]
Okay looks great.  I say just add one descriptive comment as suggested in https://github.com/OCHA-DAP/hdx-signals/pull/260#discussion_r1842295080 and go ahead and merge it so the patch is up and running on main.

We may wan't to refactor later and deal with slightly differently, but for now it should be fine. I'd say the next step which could be a small PR would just be to treat it the same as I highlighted in https://github.com/OCHA-DAP/hdx-signals/pull/260#discussion_r1842295080

this would require us to create an `iso3` code however. Perhaps we could just go with `SBA` as shown in this Wikipedia article for [Akrotiri and Dhekelia](https://en.wikipedia.org/wiki/Akrotiri_and_Dhekelia)

**Comment by martinig94** [comment]
Ah that's interesting because from [this](https://en.wikipedia.org/wiki/ISO_3166-2:CY) it looked like there is no isocode for the area. Will have a better look!

> Okay looks great. I say just add one descriptive comment as suggested in [#260 (comment)](https://github.com/OCHA-DAP/hdx-signals/pull/260#discussion_r1842295080) and go ahead and merge it so the patch is up and running on main.
> 
> We may wan't to refactor later and deal with slightly differently, but for now it should be fine. I'd say the next step which could be a small PR would just be to treat it the same as I highlighted in [#260 (comment)](https://github.com/OCHA-DAP/hdx-signals/pull/260#discussion_r1842295080)
> 
> this would require us to create an `iso3` code however. Perhaps we could just go with `SBA` as shown in this Wikipedia article for [Akrotiri and Dhekelia](https://en.wikipedia.org/wiki/Akrotiri_and_Dhekelia)

File: CHANGES.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # Changes
 
+## 0.1.14.5 (13 November 2024)
+
+### System
+
+- Add numerical iso codes to be ignored
```
[the remark this answers is not in the document]
Comment (martinig94) [reply CHANGES.md:7]: Done!

**Pull request merged**